pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
 secret_agent = None
 
 
-class NotAuthorizedException(dbus.DBusException):
-    _dbus_error_name = SECRET_AGENT_IFACE + '.NotAuthorized'
-
+class SecretAgent(object):
+    __dbus_xml__ = """
+    <node>
+        <interface name="org.freedesktop.NetworkManager.SecretAgent">
```

Comment:
I would consider to use type hints and generate the DBus API. You can check that that the expected specification is the same as the generated one in a unit test.